src/environmentd/src/http.rs

```diff
@@ -303,9 +315,41 @@ where
                 "adapter client missing".into(),
             )
         })?;
-        AuthedClient::new(&adapter_client, user.clone())
+        let mut client = AuthedClient::new(&adapter_client, user.clone())
             .await
-            .map_err(|e| (StatusCode::INTERNAL_SERVER_ERROR, e.to_string()))
+            .map_err(|e| (StatusCode::INTERNAL_SERVER_ERROR, e.to_string()))?;
+
+        // Apply options that were provided in query params.
+        let session = client.0.session();
+        let maybe_options = if params.options.is_empty() {
+            // It's possible 'options' simply wasn't provided, we don't want that to
+            // count as a failure to deserialize
+            Ok(BTreeMap::<String, String>::default())
+        } else {
+            serde_json::from_str(&params.options)
```

Comment:
Ah a json-encoded query param. My proposal had some jank in it that required url-encoding an equals sign (`?options=value%3Dkey`), but this seems better.

Reply:
Yeah I've seen the equals sign pattern in other places, but the json-encoded query param seemed to be the easiest to build on the web side, and parse on the server side.